shared/src/balancer/event_handler.rs

```diff
@@ -32,27 +38,36 @@ pub struct PoolCreated {
     pub pool_address: H160,
 }
 
-#[derive(Clone, Debug, Default, Eq, PartialEq, Hash)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub enum PoolType {
+    WeightedGeneral,
```

Comment:
What is the difference in how the services will handle a general vs a weighted two token pool? Asking, because it's not yet clear to me why we need to differentiate between the two (I assume they have different methods for swapping of fetching balances?)

Reply:
In this particular case, both types are the same as (far as I know). However, we can be certain that the other pool types will need to be differentiated for, at least, exactly the reason you mentioned (i.e. possible different transaction encoding and pool arithmetic). 

In this PR, I partially kept things separate like this mostly in preparation for what it to come. However, it is very likely that both `WeightedPool` and `WeightedPool2Tokens` don't actually need to be separated.

Reply:
How is the pool arithmetic different between general weighted pools and weighted two token pools? For swapping, I was under the impression that all we need is the poolId and the desired amounts.

Reply:
I believe I said they are not different at all. 

> In this particular case, both types are the same

> WeightedPool and WeightedPool2Tokens don't actually need to be separated.

The arithmetic is likely different for pools of **other** types.

Reply:
Yes, certainly we will want to do this. However, we have decided that at the particular moment, we will remove the pool type since these two pools don't require it and only to reintroduce this when it is absolutely necessary. With this in mind, I will sketch up a description of how each are different/handled internally when the time comes.

Reply:
From a cursory glance at the Vault code, they behave the same from the caller's point of view. The difference is only done to enable internal optimiations (2Token vs MinSwapInfo pools).

Reply:
I think ultimately we would want something like:
```
enum BalancerPool {
  Weighted(WeightedPool),
  Stable(StablePool),
  // ...
}
```

Reply:
There are only two (ish) pool types `WeightedPool` (including `TwoTokens`) and `StablePool`.

The transaction encoding is likely the same for both since we perform swaps via the Vault contract
While the arithmetic is different for each since Stable Pools use `StableMath` library while Weighted Pools uses `WeightedMath`.

This means we will need to introduce a PoolType enum (as @nlordell suggested above) down the road (i.e. when we include StablePools) that the price estimating will need to distinguish between.

Reply:
Created an issue and linked it here to track the collected information #733